Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1

```diff
@@ -9,16 +9,12 @@ Param(
 )
 
 function DownloadDependenciesFromProbingPaths($baseFolder, $project, $destinationPath) {
-    $projectSettings = ReadSettings -baseFolder $baseFolder -project $project
-    
-    $probingPaths = $projectSettings.appDependencyProbingPaths | ConvertFrom-Json
-
-    $downloadedDependencies = @()
-    if ($probingPaths) {
-        $downloadedDependencies += Get-Dependencies -probingPathsJson $repo.appDependencyProbingPaths -saveToPath $destinationPath  | Where-Object { $_ }
+    $settings = $env:Settings | ConvertFrom-Json | ConvertTo-HashTable -recurse
+    $settings = AnalyzeRepo -settings $settings -token $token -baseFolder $baseFolder -project $project -doNotCheckArtifactSetting
```

Comment:
If `AnalyzeRepo` is needed here only to determine the `appDependencyProbingPaths`'s value, consider simply creating a dedicated function to do that.

Reply:
Yep - AnalyzeRepo should be split up - I just thought that this PR is big enough to include this refactoring in that (as stated in the description)